Add object-relational mapping (orm)

gderecho opened the pull request:
Hi guys! This ORM branch is officially ready to be merged into master. It can now do all the same functions (I think) as the master branch, and I think the design has (very slightly) improved because we now don't have to manually query the tables. (Also now run with python3, not python)

Also @davenportw15 you may notice that I had a name change. I'll explain this later.

gderecho committed:
Add python3-compatible import statement

	modified:   mail.py
python3 doesn't allow relative imports,
so replace w/absolute

gderecho committed:
Add file to securely hash passwords

gderecho committed:
Modify users.py, database to use password hashes

	modified:   ../db.sqlite
Now this file uses a password hash and a salt
instead of one password field. In addition,
the auto increment feature has been turned
off because it's unneccessary

	modified:   users.py
Modify this file to use the secure crypto
hashing algorithm instead of plaintext

gderecho committed:
Modify return type of mail.py for python3

	modified:   models/mail.py
This now returns a list instead of
a filter object in python 3

gderecho committed:
Add comments to secure_pass

gderecho committed:
Raise exception if user exists

gderecho committed:
Retab mail.py

gderecho committed:
Replace with sqlalchemy orm

Not all features work yet, but the basic
functions seem to perform OK.

Also, I restructured the directory to make
it more like a python package.

You can run the application by doing
'python3 application.py'

gderecho committed:
Fix sqlite connection string

gderecho committed:
remove old files

gderecho committed:
Fix issues with usernames

Now all of the usernames
show up properly. This branch
is ready to be merged!

In addition, I removed a lot
of the old code sitting around
in comment strings.

davenportw15 commented:
This looks great. I'll familiarize myself with the changes as soon as I can. Thank you for your dedication to bringing slow email to the world.

gderecho committed:
Return error message when message isn't sent

Now it will tell the user if it can't find
the username

gderecho committed:
Fix awful security error (everyone gets all mail)

Hahahaha oops, I forgot to restrict who gets
what mail, when I checked, anyone could get
anyone else's mail!
